driver/src/main/java/org/neo4j/driver/Session.java

```diff
@@ -317,18 +318,23 @@ default void executeWriteWithoutResult( Consumer<TransactionContext> contextCons
     Result run(Query query, TransactionConfig config );
 
     /**
-     * Return the bookmark received following the last completed
-     * {@linkplain Transaction transaction}. If no bookmark was received
-     * or if this transaction was rolled back, the bookmark value will
-     * be null.
+     * Return the bookmark received following the last completed {@linkplain Transaction transaction}. If no bookmark was received or if this transaction was
+     * rolled back, the bookmark value will be null.
      *
      * @return a reference to a previous transaction
      */
+    @Deprecated
     Bookmark lastBookmark();
 
     /**
-     * Signal that you are done using this session. In the default driver usage, closing and accessing sessions is
-     * very low cost.
+     * Return an immutable set of bookmarks known by this session.
+     *
+     * @return the set of bookmarks.
+     */
+    Set<Bookmark> lastBookmarks();
```

Comment:
I am very much not happy about that change as I expressed several times (trello and standup when I remember correctly). This will break us in interesting and unnecessary ways.

Technically, using a new method name is correct. But the concept is bad.